### PR TITLE
feat(cloud_firestore_odm_generator): Support Sets as well as Lists for query method generation

### PR DESCRIPTION
**Pull request by Rexios80**
## Description

Support generating query methods for Sets.

## Related Issues

#11344

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/firebase/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/cloud_firestore_odm/cloud_firestore_odm_generator/lib/src/collection_data.dart [hunk 1]
```diff
@@ -346,7 +346,7 @@ represents the content of the collection must be in the same file.
         type.isDartCoreInt ||
         type.isDartCoreDouble ||
         type.isDartCoreBool ||
-        type.isPrimitiveList ||
+        type.isPrimitiveIterable ||
```
Comment (rrousselGit) [inline packages/cloud_firestore_odm/cloud_firestore_odm_generator/lib/src/collection_data.dart:349]: We don't want to allow any Iterable. Iterables may have an infinite length. 

We only want fixed-length collections. Such as List/Set

**Comment by rrousselGit** [comment]
This would need testing.

**Comment by rrousselGit** [comment]
This still needs testing. Could you add tests? :)

**Review by rrousselGit — CHANGES_REQUESTED** [review]
This still needs testing. Could you add tests? :)

**Comment by rrousselGit** [comment]
Sorry I might have not been specific enough as I just saw you added some things in the integration folder

In this specific scenario, we'll want to test that the generated code is correct. So we'll want a few tests that run a query using what's newly generated

File: packages/cloud_firestore_odm/cloud_firestore_odm_generator/lib/src/collection_data.dart (hunk 1 above)
Comment (rrousselGit) [inline packages/cloud_firestore_odm/cloud_firestore_odm_generator/lib/src/collection_data.dart:349]:
```suggestion
        type.isSupportedPrimitiveIterable ||
```

File: packages/cloud_firestore_odm/cloud_firestore_odm_generator/lib/src/collection_data.dart [hunk 2]
```diff
@@ -438,8 +441,12 @@ extension on DartType {
         (this as InterfaceType).typeArguments.single.isDartCoreMap;
   }
 
-  bool get isPrimitiveList {
-    if (!isDartCoreList) return false;
+  bool get isSupportedIterable =>
+      _coreListChecker.isExactlyType(this) ||
+      _coreSetChecker.isExactlyType(this);
+
+  bool get isPrimitiveIterable {
```
Comment (rrousselGit) [inline packages/cloud_firestore_odm/cloud_firestore_odm_generator/lib/src/collection_data.dart:448]:
```suggestion
  bool get isSupportedPrimitiveIterable {
```

**Review by rrousselGit — APPROVED** [review]
This is looking good! Thanks for this :)

**Review by Lyokone — APPROVED** [review]

**Pull request merged**